src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -834,6 +869,18 @@ private bool InferConnectionDetails()
                     ReportSNIError(SNIProviders.INVALID_PROV);
                     return false;
                 }
+
+                if (ConnectionProtocol == DataSource.Protocol.None)
```

Comment:
TCP protocol should not be assigned in DataSource when there is no `,` and `\`, because we don't want any postfix in SPN when only hostname is provided without protocol, `,`, and `\` in connection string.
It will fall back to TCP though because of code in `CreateConnectionHandle()`
```
switch (details.ConnectionProtocol)
{
    case DataSource.Protocol.TCP:
        sniHandle = CreateTcpHandle(details, timerExpire, callbackObject, parallel);
        break;
    case DataSource.Protocol.NP:
        sniHandle = CreateNpHandle(details, timerExpire, callbackObject, parallel);
        break;
    case DataSource.Protocol.None:
        // default to using tcp if no protocol is provided
        sniHandle = CreateTcpHandle(details, timerExpire, callbackObject, parallel);
        break;
    default:
        Debug.Fail($"Unexpected connection protocol: {details.ConnectionProtocol}");
        break;
}
```